activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/test/java/org/activiti/cloud/services/modeling/service/utils/AggregateErrorValidationStrategyTest.java

```diff
@@ -0,0 +1,115 @@
+import org.activiti.cloud.modeling.api.ValidationContext;
+import org.activiti.cloud.modeling.core.error.SemanticModelValidationException;
+import org.activiti.cloud.modeling.core.error.SyntacticModelValidationException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class AggregateErrorValidationStrategyTest {
+
+    @Mock
+    private Model model;
+
+    @Mock
+    private ModelContentValidator modelContentValidator;
+
+    @Mock
+    private ModelContentValidator otherModelContentValidator;
+
+    private final AggregateErrorValidationStrategy<ModelContentValidator> validationStrategy = new AggregateErrorValidationStrategy<>();
+
+    @Test
+    public void should_returnEmpty_whenValidatorsDoNotThrowAnyException() {
+        // when
+        assertDoesNotThrow(() -> validationStrategy.validate(List.of(modelContentValidator), validator -> {}));
+    }
+
+    @Test
+    public void should_returnAggregatedException_whenValidatorsThrowSemanticException() {
+        // given
+        final List<ModelValidationError> validatorErrors = List.of(
+            new ModelValidationError("Problem 1", "Description 1"),
+            new ModelValidationError("Problem 2", "Description 2")
+        );
+
+        final List<ModelValidationError> otherValidatorErrors = List.of(
+            new ModelValidationError("Problem 3", "Description 3")
+        );
+
+        doThrow(new SemanticModelValidationException("Test Syntactic Exception", validatorErrors))
+            .when(modelContentValidator)
+            .validateModelContent(any(), any());
+
+        doThrow(new SemanticModelValidationException("Test Syntactic Exception", otherValidatorErrors))
+            .when(otherModelContentValidator)
+            .validateModelContent(any(), any());
+
+        // when
+        SemanticModelValidationException exception = assertThrows(SemanticModelValidationException.class,
+            () -> validationStrategy.validate(
+                List.of(modelContentValidator, otherModelContentValidator),
+                validator -> validator.validateModelContent(model.getContent(), ValidationContext.EMPTY_CONTEXT)
+            ));
+
+        // then
+        assertThat(exception).hasMessage("Semantic model validation errors encountered: 3 schema violations found");
+        assertThat(exception.getValidationErrors()).isNotNull().hasSize(3).containsExactly(Stream.concat(validatorErrors.stream(),
+                otherValidatorErrors.stream()).collect(Collectors.toList()).toArray(ModelValidationError[]::new));
+    }
+
+    @Test
+    public void should_returnException_whenAValidatorDoesNotThrowSemanticException() {
```

Comment:
Naming: `*whenAValidatorDoesNotThrowSemanticException` --> `*whenAValidatorThrowsSemanticException` ?

Reply:
Renamed to `*whenAValidatorThrowsSyntacticException` since it tests that the strategy is not changing the validation behavior for other types of error.